### PR TITLE
Autovacuum crypto_transfer, token_transfer, and transaction

### LINKED ISSUE #1780: Querying REST api accounts endpoint for 0.0.9 times out in production environment
**Detailed Description**
The query `/api/v1/accounts/0.0.9` times out in production environment.

pg planner chose to use index `crypto_transfer__consensus_timestamp` to do a index backwards scan:

1. retrieving fields (consensus_timestamp, entity_id) needs to two visits: one to the index and one to the actual row
2. more than 50 million rows are skipped to hit the first match

This particular subquery takes tens of seconds to finish. pg planner does not choose the index on (entity_id, consensus_timestamp) for much faster index only scan because the visibility map info for the table is outdated.

**Actual Behavior**

**Expected Behavior**
The aforementioned query should return results timely, ideally within a second or better.

**Environment:**

**Additional Context**

### LINKED ISSUE #1849: vacuum error in k8s with timescaledb
**Detailed Description**
Hit the following issue when trying to manually vacuum table `account_balance`:

```
mirror_node=> vacuum VERBOSE account_balance;
INFO:  vacuuming "_timescaledb_internal._hyper_1_21_chunk"
INFO:  launched 1 parallel vacuum worker for index cleanup (planned: 1)
INFO:  "_hyper_1_21_chunk": found 0 removable, 48 nonremovable row versions in 1 out of 33535 pages
DETAIL:  0 dead row versions cannot be removed yet, oldest xmin: 30744
There were 0 unused item identifiers.
Skipped 0 pages due to buffer pins, 0 frozen pages.
0 pages are entirely empty.
CPU: user: 0.02 s, system: 0.08 s, elapsed: 0.13 s.
INFO:  vacuuming "_timescaledb_internal._hyper_1_27_chunk"
ERROR:  could not resize shared memory segment "/PostgreSQL.1492849781" to 110939168 bytes: No space left on device
```

shm size shown in the `timescale` container of pod `dev-timescale-1` (the primary in the release):

```
postgres@dev-timescaledb-1:~$ df -h
Filesystem      Size  Used Avail Use% Mounted on
shm              64M   20K   64M   1% /dev/shm
```

64M is the default shared memory size, and there is a way to increase it:

https://stackoverflow.com/questions/46085748/define-size-for-dev-shm-on-container-engine/46434614#46434614

**Actual Behavior**

**Expected Behavior**
vacuum should succeed

**Environment:**
- chart version: 0.18.0-rc1
- timescaledb image version: timescale/timescaledb-ha:pg13-ts2.2-latest

**Additional Context**

### PR DESCRIPTION
**Pull request by xin-hedera**
**Detailed description**:

- configure more frequent anti-wraparound autovacuum for `crypto_transfer`, `token_transfer`, and `transaction` for pg 9.6
- configure more frequent insert-only autovacuum for `crypto_transfer`, `token_transfer`, and `transaction` for pg 13
- upgrade to timescale/timescaledb-ha:pg13-ts2.2-latest
- increase shm size for timescaledb container

**Which issue(s) this PR fixes**:
Fixes #1780
Fixes #1849 

**Special notes for your reviewer**:

Criteria for the tables to enable more frequent autovacuum:

- insert-only
- there are sql queries on the table which can benefit from index-only scan

***For pg 9.6***

Reference doc: https://www.postgresql.org/docs/9.6/routine-vacuuming.html section 24.1.5

- `autovacuum_freeze_max_age` can't be set lower than 100,000, mainnet postgresql tx id grows at around 60,000/day
- setting `autovacuum_freeze_table_age` is needed otherwise postgresql will use the global parameter `vacuum_freeze_table_age` which is 150,000,000. then the autovacuum ends up not freezing any pages and running forever since the table age won't change
- `autovacuum_freeze_min_age = 0` is aggressive, freezes all pages at the time of acutovacuum scan
- `log_autovacuum_min_duration = 0` turns on logging of autovacuum of the table

Concerns:

- anti-wraparound autovacuum scan will lock the table from DDL, so this will be a problem for ops doing upgrades

***For timescaledb with pg13***

pg 13 introduces a new method to configure autovacuum for insert-only tables:

1. `autovacuum_vacuum_insert_threshold`, default to 1000
2. `autovacuum_vacuum_insert_scale_factor`, default to 0.2

autovacuum will run when the number of inserts exceeds `autovacuum_vacuum_insert_threshold + reltuples * autovacuum_vacuum_insert_scale_factor` for a table since last autovacuum. `reltuples` is the approximate table size.

The values set for table `crypto_transfer` and `transaction` are based on the current daily transaction volume in mainnet. `token_transfer` has very few transactions (less than 6k in the past week) so the value will not have a real impact on performance and we should adjust it when the traffic grows.  

For timescaledb, autovacuum will only scan the current chunk, which is more efficient.

**Checklist**
- [ ] Documentation added
- [ ] Tests updated

### CODE REVIEW

**Review by steven-sheehy — COMMENTED** [review]
Looks good. Do we need to consider doing account_balance as well?

**Comment by xin-hedera** [comment]
> Looks good. Do we need to consider doing account_balance as well?

Checked `account_balance` table and `token_balance` table, the queries on the two tables either just select the consensus_timestamp (look for the latest consensus_timestamp or the latest at certain time in the past) or select the full row.

The consensus_timestamp query will not benefit much from index only scan since we just look for one consensus_timestamp and at most pg just has to spend an extra trip to make sure the corresponding index's data in the page is not stale.

The other queries always select the balance and balance is not in any indexes, so they are not index-only scans.

So vacuuming these tables more frequently won't speed up our queries thus we shouldn't do it.

File: hedera-mirror-importer/src/main/resources/db/migration/v2/V2.0.5__autovacuum_insert_only_tables.sql [hunk 1]
```diff
@@ -0,0 +1,25 @@
+-------------------
+-- autovacuum insert-only tables more frequently to ensure most pages are visible for index-only scans
+-- prior to postgresql 13, this can be achieved by configuring a more aggressive anti-wraparound autovacuum
+-------------------
+
```
Comment (Nana-EC) [inline hedera-mirror-importer/src/main/resources/db/migration/v2/V2.0.5__autovacuum_insert_only_tables.sql:5]: nit: 
```sql
\set autovacuum_freeze_max_age '100000'
\set autovacuum_freeze_table_age '100000'
```

Or if there's value to configurability you can add it as property in the `application.yaml` under `spring.flyway.placeholders`

**Review by Nana-EC — COMMENTED** [review]
Looks good, nit suggestion based on your preference.
Also is transactions not a huge insert only table.
Would some queries such as account and time range benefit from a vacuum more often?

**Review by steven-sheehy — APPROVED** [review]

**Comment by steven-sheehy** [comment]
We'll be bumping our TimescaleDB chart to use PostgreSQL 13 [soon](https://github.com/timescale/timescaledb-kubernetes/pull/283) and our timescale.com SaaS uses PostgreSQL 13. Can you update the schema v2 stuff to use the newer pg13 approach you mention?

**Comment by xin-hedera** [comment]
> We'll be bumping our TimescaleDB chart to use PostgreSQL 13 [soon](https://github.com/timescale/timescaledb-kubernetes/pull/283) and our timescale.com SaaS uses PostgreSQL 13. Can you update the schema v2 stuff to use the newer pg13 approach you mention?

definitely. the pg13 new insert-only table autovacuum mechanism will be more efficient.

**Review by ijungmann — APPROVED** [review]

File: hedera-mirror-importer/src/main/resources/db/migration/v2/V2.0.5__autovacuum_insert_only_tables.sql (hunk 1 above)
Comment (xin-hedera) [reply hedera-mirror-importer/src/main/resources/db/migration/v2/V2.0.5__autovacuum_insert_only_tables.sql]: changed both to repeatable migrations and added properties to spring.flyway.placeholders

**Review by Nana-EC — DISMISSED** [review]
LGTM

File: hedera-mirror-grpc/src/test/resources/config/application.yml [hunk 2]
```diff
@@ -40,6 +40,10 @@ spring:
       chunkIdInterval: 10000
       chunkTimeInterval: 604800000000000
       compressionAge: 9223372036854775807 # use long max to avoid compression during test
+      insertOnlyAutovacuumFreezeMaxAge: 100000
```
Comment (steven-sheehy) [inline hedera-mirror-grpc/src/test/resources/config/application.yml:43]: Does it work if we remove these placeholders in test config? Would prefer to not duplicate them if possible.

File: hedera-mirror-importer/src/main/resources/application.yml [hunk 3]
```diff
@@ -86,6 +86,10 @@ spring:
       chunkIdInterval: 10000
       chunkTimeInterval: 604800000000000
       compressionAge: 604800000000000
+      insertOnlyAutovacuumFreezeMaxAge: 100000
```
Comment (steven-sheehy) [inline hedera-mirror-importer/src/main/resources/application.yml:89]: Would prefer all variables start with `autovacuum` so we can group them. The name should match the underlying postgres property prefix but in camel case form. Also sorted would be nice.

**Review by steven-sheehy — CHANGES_REQUESTED** [review]

File: hedera-mirror-importer/src/main/resources/application.yml (hunk 3 above)
Comment (xin-hedera) [reply hedera-mirror-importer/src/main/resources/application.yml:89]: sure

**Review by ijungmann — DISMISSED** [review]

File: hedera-mirror-grpc/src/test/resources/config/application.yml (hunk 2 above)
Comment (steven-sheehy) [reply hedera-mirror-grpc/src/test/resources/config/application.yml:43]: That's fine, let's keep as is.

**Review by steven-sheehy — APPROVED** [review]
LGTM

**Review by Nana-EC — APPROVED** [review]
LGTM

**Pull request merged**